### DISCUSSION TITLE
Help creating access permissions on routes <Link state={ customItem: 'newItem'}

### DISCUSSION THREAD

**Discussion by efgomes** [TanStack/router#1865] [Q&A] [answered]
![image](https://github.com/TanStack/router/assets/53862865/e2a1375c-f11b-4149-b554-16768b6caa0e)

I'm passing a new attribute to Link through state, which was the way I found to identify which module this route belongs to.

![image](https://github.com/TanStack/router/assets/53862865/e4a08e09-5b46-44a9-891d-e3975baf0bf5)

it's working

But I would like to know if this is the correct way to do this?

Another question, how can I resolve this typoscript error?
![image](https://github.com/TanStack/router/assets/53862865/94e8b120-6dee-433b-9f7a-7fe0a77a3837)

**Comment by schiller-manuel** [comment, marked as answer]
you can use it, but you could also use search params.
about that TS error, have a look at https://github.com/TanStack/router/discussions/1342#discussioncomment-8848490
